packages/shared/lib/core/profile-manager/api/events/transaction-progress.event.ts

```diff
@@ -10,42 +9,54 @@ import {
 } from '@core/profile-manager'
 import { isOnboardingLedgerProfile } from '@contexts/onboarding'
 import { closePopup, openPopup } from '@lib/popup'
-
 import { TransactionProgressEventPayload } from '../types'
+import { deconstructLedgerVerificationProps } from '@core/ledger/helpers'
 
 export function handleTransactionProgressEvent(accountId: string, payload: TransactionProgressEventPayload): void {
     if (get(isActiveLedgerProfile)) {
         if (get(selectedAccountId) === accountId) {
-            handleTransactionProgressInternal(payload)
+            TransactionProgressEventForLedger(payload)
         }
     } else if (get(isOnboardingLedgerProfile)) {
-        handleTransactionProgressInternal(payload, true)
+        TransactionProgressEventForLedger(payload)
     } else {
         console.warn('Transaction progress handler unimplemented: ', payload)
     }
 }
 
-function handleTransactionProgressInternal(
-    payload: TransactionProgressEventPayload,
-    isDuringOnboarding?: boolean
-): void {
+function TransactionProgressEventForLedger(payload: TransactionProgressEventPayload): void {
     if (payload) {
-        if (payload === 'PerformingPow') {
-            /**
-             * NOTE: This is necessary so that the transaction
-             * confirmation popup is closed since we can assume
-             * the user confirmed the prompt on the actual Ledger
-             * device.
-             */
-            closePopup(true)
-        } else if (isPreparedTransaction(payload)) {
-            openLedgerConfirmationPopup(payload, isDuringOnboarding)
+        if (isPreparedTransaction(payload)) {
+            openPopup({
+                type: 'verifyLedgerTransaction',
+                hideClose: true,
+                preventClose: true,
+                props: {
+                    ...deconstructLedgerVerificationProps(),
+                },
+            })
         } else if (isPreparedTransactionEssenceHash(payload)) {
             if (get(ledgerNanoStatus)?.blindSigningEnabled) {
-                openLedgerConfirmationPopup(payload, isDuringOnboarding)
+                openPopup({
+                    type: 'verifyLedgerTransaction',
+                    hideClose: true,
+                    preventClose: true,
+                    props: {
+                        hash: payload?.['PreparedTransactionEssenceHash'],
+                    },
+                })
             } else {
-                openPopup({ type: 'enableLedgerBlindSigning', hideClose: true, preventClose: true })
+                openPopup({
+                    type: 'enableLedgerBlindSigning',
+                    hideClose: true,
+                    preventClose: true,
+                    props: {
+                        hash: payload?.['PreparedTransactionEssenceHash'],
+                    },
```

Comment:
Not sure if we need to pass the hash here since it's the enable blind signing instructions popup.